src/Listener/NoCacheImageRequestSubscriber.php

```diff
@@ -48,10 +48,10 @@ public function onRequest(RequestEvent $requestEvent): void
                 );
             } else {
                 $response = new Response(
-                    (string) $image->encode(null, $requestEvent->getQuality()),
+                    $image->encode()->toString(),
                     Response::HTTP_OK,
                     [
-                        'Content-Type' => $image->mime(),
+                        'Content-Type' => $image->encode()->mimetype(),
```

Comment:
You should **not** reencode image just to get its mimetype!

Extract a locale variable here.